### PR TITLE
Mostly split and named splice TUs and matched several functions

### PR DESCRIPTION
**Pull request by Yotona**

### CODE REVIEW

File: src/P2/splice/spliceutils.cpp [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+//     g_splotheapMethod.Startup(0xc,0x80);
+// }
 
-INCLUDE_ASM(const s32, "P2/splice/spliceutils", ShutdownSplice__Fv);
+void ShutdownSplice(void) {
+    g_splotheapPair.Shutdown();
+    g_splotheapFrame.Shutdown();
+    g_splotheapUnk1.Shutdown();
+    g_splotheapProc.Shutdown();
+    ShutdownSpliceStructuredTypeFactories();
```
Comment (TheOnlyZac) [inline src/P2/splice/spliceutils.cpp:29]: Missing indent

File: src/P2/splice/proc.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    pprocClone->m_ppairCodeExpr = m_ppairCodeExpr;
+}
+
+// todo: matches but blocked by references in CRef::CloneTo and RefEvalLambda
+INCLUDE_ASM(const s32, "P2/splice/proc", PprocNew__Fv);
+// static CProc* PprocNew(void) {
+//     CProc* proc = (CProc*)g_splotheapProc.PvAllocClear();
+//     memset(proc, 0, sizeof(CProc));
+//     return proc;
+// }
```
Comment (TheOnlyZac) [inline src/P2/splice/proc.cpp:23]: Missing new line at EOF

**Review by TheOnlyZac — APPROVED** [review]
I will need to confirm it builds on my machine but it looks great, thanks for doing all this! Just a few style nitpicks, but nothing that would prevent me merging this.

**Comment by TheOnlyZac** [comment]
@Yotona Does this build for you? I am getting this output from ninja:
```
[5/168] cc src/P2/splice/eval.cpp
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
[6/168] cc src/P2/splice/gc.cpp
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
[7/168] cc src/P2/splice/method.cpp
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
[8/168] cc src/P2/splice/pair.cpp
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
[9/168] cc src/P2/splice/ref.cpp
FAILED: obj/src/P2/splice/ref.cpp.o 
wine /workspaces/sly1/tools/cc/bin/ee-gcc.exe -c -Iinclude -isystem include/sdk/ee -isystem include/gcc -x c++ -V 2.95.2 -O2 -G0 -ffast-math src/P2/splice/ref.cpp -o obj/src/P2/splice/ref.cpp.o && mips-linux-gnu-strip obj/src/P2/splice/ref.cpp.o -N dummy-symbol-name
src/P2/splice/ref.cpp:3: string.h: No such file or directory
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
[10/168] cc src/P2/splice/proc.cpp
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
[11/168] cc src/P2/splice/bif.cpp
ee-gcc.exe: file path prefix `/usr/local/sce/ee/gcc/lib/gcc-lib/ee\2.95.2\' never used
ninja: build stopped: subcommand failed.
```
Error is `src/P2/splice/ref.cpp:3: string.h: No such file or directory`

**Comment by Yotona** [comment]
Yes locally it builds, looks like it was an issue with an include path to access memcmp. Believe that comes from the libc implementation that sce utilizes, I can try to stub it in somewhere. Not technically even necessary while the function that calls it is non-matched

**Comment by TheOnlyZac** [comment]
Thank you again! I think this is going to bump us up by nearly 0.1% 👏🏻

**Pull request merged**